snapcraft/plugins/ruby.py

```diff
@@ -0,0 +1,144 @@
+
+        self._ruby_version = self.options.ruby_version
+        self._ruby_part_dir = join(self.partdir, 'ruby')
+        self._ruby_download_url = \
+            'https://cache.ruby-lang.org/pub/ruby/ruby-{}.tar.gz'.format(
+                self._ruby_version)
+        self._ruby_tar = Tar(self._ruby_download_url, self._ruby_part_dir)
+        self._gems = self.options.gems
+        self._install_bundler = False
+
+        self.build_packages.extend(['gcc', 'g++', 'make', 'zlib1g-dev',
+                                    'libssl-dev', 'libreadline-dev'])
+
+        version_map = [('2.%d.[0-9]' % i, '2.%d.0' % i) for i in range(5)]
+
+        for version_regex, version_dir in version_map:
+            if re.compile(version_regex).match(self._ruby_version):
+                self._ruby_version_dir = version_dir
+                break
+
+    def pull(self):
+        super().pull()
+        makedirs(self._ruby_part_dir, exist_ok=True)
+        self._ruby_tar.download()
+
+    def build(self):
+        super().build()
+        self._ruby_install(builddir=self._ruby_part_dir)
+        self._gem_install()
+        if self._install_bundler:
+            self._bundle_install()
+
+    def env(self, root):
+        env = {}
+        env['PATH'] = '{}:{}'.format(join(root, 'bin'), environ['PATH'])
+        env['RUBYPATH'] = '{}'.format(join(root, 'bin'))
+        rubydir = join(root, 'lib', 'ruby')
+        rubylib = join(rubydir, self._ruby_version_dir)
+        env['RUBYLIB'] = '{}:{}'.format(
+            rubylib, join(rubylib, '{}-linux'.format(platform.machine())))
+        env['GEM_HOME'] = join(rubydir, 'gems', self._ruby_version_dir)
+        env['GEM_PATH'] = join(rubydir, 'gems', self._ruby_version_dir)
+        return env
+
+    def _ruby_install(self, builddir):
+        self._ruby_tar.provision(
+            builddir, clean_target=False, keep_tarball=True)
+        self.run(['./configure', '--disable-install-rdoc', '--prefix=/'],
+                 cwd=builddir)
+        self.run(['make', '-j{}'.format(self.parallel_build_count)],
+                 cwd=builddir)
+        self.run(['make', 'install', 'DESTDIR={}'.format(self.installdir)],
+                 cwd=builddir)
+
+    def _gem_install(self):
+        if exists(join(self.builddir, 'Gemfile')):
+            self._install_bundler = True
+            self._gems = self._gems + ['bundler']
+        if self._gems:
+            gem_install_cmd = [join(self.installdir, 'bin', 'ruby'),
+                               join(self.installdir, 'bin', 'gem'), 'install']
+            self.run(gem_install_cmd + self._gems,
+                     env=self.env(root=self.installdir))
+
+    def _bundle_install(self):
+        bundle_install_cmd = [join(self.installdir, 'bin', 'ruby'),
+                              join(self.installdir, 'bin', 'bundle'),
+                              'install']
+        self.run(bundle_install_cmd, env=self.env(root=self.installdir))
```

Comment:
@kyrofa @kalikiana Thanks for your feedback here. Looking into these things now.